datadog_checks_base/datadog_checks/base/utils/tls.py

```diff
@@ -109,13 +109,13 @@ def _create_tls_context(self):
         context = ssl.SSLContext(protocol=ssl.PROTOCOL_TLS)
```

Comment:
Note: one of the difference here is that protocol changed from `PROTOCOL_TLS_CLIENT` to `PROTOCOL_TLS`, which changes default value for `check_hostname` and `verify_mode`, and we don't re-update `check_hostname` when `tls_verify=False`
https://docs.python.org/3/library/ssl.html#ssl.PROTOCOL_TLS_CLIENT
```
Auto-negotiate the highest protocol version that both the client and server support, and configure the context client-side connections. The protocol enables CERT_REQUIRED and check_hostname by default.
```

https://github.com/DataDog/integrations-core/blob/a03c2440c59487e3c8c0ef39d54dab23deb83afe/tls/datadog_checks/tls/tls.py#L343